cloud/ec2_fleet.go

```diff
@@ -0,0 +1,538 @@
+package cloud
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mitchellh/mapstructure"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+type ec2FleetManager struct {
+	client      AWSClient
+	credentials *credentials.Credentials
+	settings    *evergreen.Settings
+}
+
+func (m *ec2FleetManager) GetSettings() ProviderSettings {
+	return &EC2ProviderSettings{}
+}
+
+func (m *ec2FleetManager) Configure(ctx context.Context, settings *evergreen.Settings) error {
+	if m.client == nil {
+		m.client = &awsClientImpl{}
```

Comment:
looking at the client impl, it looks like we need more configuration (e.g. a nil http client/etc.) is there a case where you create a manager without a client?

Reply:
removed.